src/main/scala/org/broadinstitute/dsde/workbench/leonardo/dao/GoogleDataprocDAO.scala

```diff
@@ -120,39 +122,28 @@ class GoogleDataprocDAO(protected val dataprocConfig: DataprocConfig, protected
     }
   }
 
-  private def getFirewallRule(googleProject: String, firewallRuleName: String): Future[Unit] = {
-    Future {
-      val request = compute.firewalls().get(googleProject, firewallRuleName)
-      executeGoogleRequest(request) // returns a Firewall
-      ()
-    }.recoverWith {
-      case e: GoogleJsonResponseException =>
-        if (e.getStatusCode == 404)
-          throw FirewallRuleNotFoundException(googleProject, firewallRuleName)
-        throw CallToGoogleApiFailedException(googleProject, firewallRuleName, e.getStatusCode, e.getDetails.getMessage)
-    }
+  private def getFirewallRule(googleProject: GoogleProject, firewallRuleName: String): Future[Unit] = {
+    val request = compute.firewalls().get(googleProject, firewallRuleName)
+    executeGoogleRequestAsync(googleProject, firewallRuleName, request).recover {
+      case CallToGoogleApiFailedException(_, _, 404, _) =>
+        // throw FirewallRuleNotFoundException in case of 404 errors
+        throw FirewallRuleNotFoundException(googleProject, firewallRuleName)
+    }.void
```

Comment:
`.void` converts any `Future[A]` into a `Future[Unit]`. 

(`.mapTo[Unit]` doesn't work because it does a cast; `.void` works because it essentially does a `.map(_ => ())` but looks cleaner :))

Reply:
Great, I like using `void` better than `()`!

Reply:
Yeah, it's a cats-ism. Any Functor has a `.void`. :)